packages/phpstan-rules/src/Rules/NarrowType/NarrowPublicClassMethodParamTypeByCallerTypeRule.php

```diff
@@ -65,6 +65,10 @@ public function processNode(Node $node, Scope $scope): array
                     continue;
                 }
 
+                if (is_a($uniqueCollectedArgTypesString, $paramTypesString,true)) {
```

Comment:
this possibly make false positive that make no longer narrowed, probably better to check variable name, I will verify more...

Reply:
I think show the error message is correct, but remove the "$this()" in the message:

```diff
-'11: Parameters should use "$this(Symplify\PHPStanRules\Tests\Rules\NarrowType\NarrowPublicClassMethodParamTypeByCallerTypeRule\Source\ExpectedThisType\CallByThis)" types as the only types passed to this method
+'11: Parameters should use "Symplify\PHPStanRules\Tests\Rules\NarrowType\NarrowPublicClassMethodParamTypeByCallerTypeRule\Source\ExpectedThisType\CallByThis" types as the only types passed to this method
```

Reply:
fixed by skip $this pass exact type.